SingletonPattern/SingletonPattern.cpp

```diff
@@ -3,25 +3,30 @@ using namespace std;
 
 class Singleton {
 public:
-    static Singleton* Instance() {
-        if (!_instance) {
-            _instance = new Singleton();
-        }
-        return _instance;
+    static Singleton& Instance() {
+        static Singleton instance;
+        return instance;
     }
+
+    void printInstanceAddr() {
+        cout << this << endl;
+    }
+
 private:
-    Singleton() { cout << "Singleton..." << endl; }
-    static Singleton* _instance;
+    Singleton() {
+        cout << "ctor" << endl;
+    }
+
+    Singleton(const Singleton &another) {}
 };
-Singleton* Singleton::_instance = nullptr;
+
 
 int main() {
-    Singleton *singleton1 = Singleton::Instance();
-    cout << singleton1 << endl;
-    Singleton *singleton2 = Singleton::Instance();
-    cout << singleton2 << endl;
+    Singleton &singleton1 = Singleton::Instance();
+    singleton1.printInstanceAddr();
```

Comment:
`printInstanceAddr()` is wordy, why not `cout << &singleton1 << endl;` directly?

Reply:
Silly me.
I will rewrite it.